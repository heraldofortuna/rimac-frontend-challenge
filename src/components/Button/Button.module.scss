.button {
  background-color: $grey-100-color;
  width: 100%;
  color: $grey-0-color;
  font-size: 18px;
  font-weight: 700;
  line-height: 20px;
  letter-spacing: 0.4px;
  padding: 18px;
  border: 0;
  border-radius: 40px;
  cursor: pointer;

  @media (min-width: $breakpoint-desktop) {
    width: fit-content;
    font-size: 20px;
    line-height: 24px;
    padding: 20px 40px;
  }
}