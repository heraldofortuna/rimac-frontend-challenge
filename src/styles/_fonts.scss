@font-face {
  font-family: 'BR Sonoma Regular';
  font-style: normal;
  font-weight: normal;
  src: local('BR Sonoma Regular'), url('/fonts/BRSonoma-Regular-BF654c45266c042.woff') format('woff');
}
  

@font-face {
  font-family: 'BR Sonoma Regular Italic';
  font-style: normal;
  font-weight: normal;
  src: local('BR Sonoma Regular Italic'), url('/fonts/BRSonoma-RegularItalic-BF654c452681c11.woff') format('woff');
}
  

@font-face {
  font-family: 'BR Sonoma Thin';
  font-style: normal;
  font-weight: normal;
  src: local('BR Sonoma Thin'), url('/fonts/BRSonoma-Thin-BF654c45255ffe0.woff') format('woff');
}
  

@font-face {
  font-family: 'BR Sonoma Thin Italic';
  font-style: normal;
  font-weight: normal;
  src: local('BR Sonoma Thin Italic'), url('/fonts/BRSonoma-ThinItalic-BF654c45268d3f9.woff') format('woff');
}
  

@font-face {
  font-family: 'BR Sonoma ExtraLight';
  font-style: normal;
  font-weight: normal;
  src: local('BR Sonoma ExtraLight'), url('/fonts/BRSonoma-ExtraLight-BF654c45265af8d.woff') format('woff');
}
  

@font-face {
  font-family: 'BR Sonoma ExtraLight Italic';
  font-style: normal;
  font-weight: normal;
  src: local('BR Sonoma ExtraLight Italic'), url('/fonts/BRSonoma-ExtraLightItalic-BF654c4525a5046.woff') format('woff');
}
  

@font-face {
  font-family: 'BR Sonoma Light';
  font-style: normal;
  font-weight: normal;
  src: local('BR Sonoma Light'), url('/fonts/BRSonoma-Light-BF654c452608e0f.woff') format('woff');
}
  

@font-face {
  font-family: 'BR Sonoma Light Italic';
  font-style: normal;
  font-weight: normal;
  src: local('BR Sonoma Light Italic'), url('/fonts/BRSonoma-LightItalic-BF654c45266aa83.woff') format('woff');
}
  

@font-face {
  font-family: 'BR Sonoma Medium';
  font-style: normal;
  font-weight: normal;
  src: local('BR Sonoma Medium'), url('/fonts/BRSonoma-Medium-BF654c45266edd1.woff') format('woff');
}
  

@font-face {
  font-family: 'BR Sonoma Medium Italic';
  font-style: normal;
  font-weight: normal;
  src: local('BR Sonoma Medium Italic'), url('/fonts/BRSonoma-MediumItalic-BF654c45267d45f.woff') format('woff');
}
  

@font-face {
  font-family: 'BR Sonoma SemiBold';
  font-style: normal;
  font-weight: normal;
  src: local('BR Sonoma SemiBold'), url('/fonts/BRSonoma-SemiBold-BF654c45268c340.woff') format('woff');
}
  

@font-face {
  font-family: 'BR Sonoma SemiBold Italic';
  font-style: normal;
  font-weight: normal;
  src: local('BR Sonoma SemiBold Italic'), url('/fonts/BRSonoma-SemiBoldItalic-BF654c452696350.woff') format('woff');
}
  

@font-face {
  font-family: 'BR Sonoma Bold';
  font-style: normal;
  font-weight: normal;
  src: local('BR Sonoma Bold'), url('/fonts/BRSonoma-Bold-BF654c4526823f5.woff') format('woff');
}
  

@font-face {
  font-family: 'BR Sonoma Bold Italic';
  font-style: normal;
  font-weight: normal;
  src: local('BR Sonoma Bold Italic'), url('/fonts/BRSonoma-BoldItalic-BF654c4525c9c27.woff') format('woff');
}
  

@font-face {
  font-family: 'BR Sonoma Black';
  font-style: normal;
  font-weight: normal;
  src: local('BR Sonoma Black'), url('/fonts/BRSonoma-Black-BF654c4525506bf.woff') format('woff');
}
  

@font-face {
  font-family: 'BR Sonoma Black Italic';
  font-style: normal;
  font-weight: normal;
  src: local('BR Sonoma Black Italic'), url('/fonts/BRSonoma-BlackItalic-BF654c45268988e.woff') format('woff');
}