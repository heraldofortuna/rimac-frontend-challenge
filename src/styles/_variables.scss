$grey-100-color: #03050F;
$grey-80-color: #2B304E;
$grey-30-color: #CCD1EE;
$grey-10-color: #F8F9FF;
$blue-berry-600-color: #4F4FFF;
$neutrals-700-color: #141938;
$neutrals-600-color: #7981B2;
$neutrals-0-color: #FFFFFF;

$sonoma-font: 'BR Sonoma Regular', sans-serif;
$lato-font: 'Lato', sans-serif;

$breakpoint-mobile: 480px;
$breakpoint-desktop: 768px;