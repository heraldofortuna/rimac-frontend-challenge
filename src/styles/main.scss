@use 'fonts';
@import url('https://fonts.googleapis.com/css2?family=Lato:wght@300;400;700;900&display=swap');

* {
  padding: 0;
  margin: 0;
}

body {
  background-color: $grey-10-color;
  font-family: $sonoma-font;
  color: $grey-100-color;
}

#root {
  min-height: 100vh;
  display: flex;
  flex-direction: column;
}