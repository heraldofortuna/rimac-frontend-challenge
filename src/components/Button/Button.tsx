import type { IButtonProps } from '../../types/components/button';
import styles from './Button.module.scss';

const Button: React.FC<IButtonProps> = ({
  type = "button",
  text,
  size = "medium",
  color = "black",
  isDisabled = false,
  isLoading = false,
  loadingText = "Cargando",
  onClick = () => {}
}) => {
  const buttonClass = `${styles.button} ${styles[size]} ${styles[color]} ${
    isDisabled ? styles.disabled : ''
  } ${isLoading ? styles.loading : ''}`;
  
  return (
    <button 
      type={type} 
      disabled={isDisabled || isLoading}
      onClick={isDisabled || isLoading ? undefined : onClick} 
      className={buttonClass}
    >
      {isLoading ? (
        <>
          <span className={styles.loader}></span>
          {loadingText}
        </>
      ) : (
        text
      )}
    </button>
  )
}

export default Button;