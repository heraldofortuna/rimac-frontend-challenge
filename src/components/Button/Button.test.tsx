import { describe, it, expect, vi } from 'vitest';
import { render, screen, fireEvent } from '@testing-library/react';
import Button from './Button';
import type { IButtonProps } from "../../types/components/button";
import styles from "./Button.module.scss";

describe('Button Component', () => {
  const defaultProps: IButtonProps = {
    text: 'Test Button',
    onClick: vi.fn(),
  };

  const renderButton = (props: Partial<IButtonProps> = {}) => {
    return render(<Button {...defaultProps} {...props} />);
  };

  it('renders correctly with default props', () => {
    renderButton();
    
    const button = screen.getByRole('button', { name: 'Test Button' });
    expect(button).toBeInTheDocument();
    expect(button).toHaveAttribute('type', 'button');
    expect(button).toHaveClass(styles.button);
    expect(button).toHaveClass(styles.medium);
    expect(button).toHaveClass(styles.black);
    expect(button).not.toBeDisabled();
  });

  it('applies correct size and color classes', () => {
    renderButton({ size: 'small', color: 'red' });
    
    const button = screen.getByRole('button');
    expect(button).toHaveClass(styles.small);
    expect(button).toHaveClass(styles.red);
    expect(button).not.toHaveClass(styles.medium);
    expect(button).not.toHaveClass(styles.black);
  });

  it('calls onClick handler when clicked', () => {
    const handleClick = vi.fn();
    renderButton({ onClick: handleClick });
    
    const button = screen.getByRole('button');
    fireEvent.click(button);
    
    expect(handleClick).toHaveBeenCalledTimes(1);
  });

  it('does not call onClick when disabled', () => {
    const handleClick = vi.fn();
    renderButton({ isDisabled: true, onClick: handleClick });
    
    const button = screen.getByRole('button');
    fireEvent.click(button);
    
    expect(handleClick).not.toHaveBeenCalled();
    expect(button).toBeDisabled();
  });

  it('does not call onClick when loading', () => {
    const handleClick = vi.fn();
    renderButton({ isLoading: true, onClick: handleClick });
    
    const button = screen.getByRole('button');
    fireEvent.click(button);
    
    expect(handleClick).not.toHaveBeenCalled();
    expect(button).toBeDisabled();
  });

  it('shows loading state correctly', () => {
    renderButton({ isLoading: true, loadingText: 'Please wait...' });
    
    const button = screen.getByRole('button');
    expect(button).toHaveTextContent('Please wait...');
    expect(button).toHaveClass(styles.loading);
    expect(screen.getByTestId('button-loader')).toBeInTheDocument();
  });

  it('applies disabled styles when disabled', () => {
    renderButton({ isDisabled: true });
    
    const button = screen.getByRole('button');
    expect(button).toHaveClass(styles.disabled);
    expect(button).toBeDisabled();
  });

  it('matches snapshot with default props', () => {
    const { asFragment } = renderButton();
    expect(asFragment()).toMatchSnapshot();
  });

  it('matches snapshot when loading', () => {
    const { asFragment } = renderButton({ isLoading: true });
    expect(asFragment()).toMatchSnapshot();
  });

  it('matches snapshot when disabled', () => {
    const { asFragment } = renderButton({ isDisabled: true });
    expect(asFragment()).toMatchSnapshot();
  });
});