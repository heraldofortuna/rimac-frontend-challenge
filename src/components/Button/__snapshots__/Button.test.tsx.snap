// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button Component > matches snapshot when disabled 1`] = `
<DocumentFragment>
  <button
    class="_button_ee73f7 _medium_ee73f7 _black_ee73f7 _disabled_ee73f7 "
    disabled=""
    type="button"
  >
    Test Button
  </button>
</DocumentFragment>
`;

exports[`Button Component > matches snapshot when loading 1`] = `
<DocumentFragment>
  <button
    class="_button_ee73f7 _medium_ee73f7 _black_ee73f7  _loading_ee73f7"
    disabled=""
    type="button"
  >
    <span
      class="_loader_ee73f7"
      data-testid="button-loader"
    />
    Cargando
  </button>
</DocumentFragment>
`;

exports[`Button Component > matches snapshot with default props 1`] = `
<DocumentFragment>
  <button
    class="_button_ee73f7 _medium_ee73f7 _black_ee73f7  "
    type="button"
  >
    Test Button
  </button>
</DocumentFragment>
`;
